examples/platform/silabs/SiWx917/SiWx917/hal/rsi_hal_mcu_m4.c

```diff
@@ -0,0 +1,303 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "rsi_board.h"
+#include "rsi_chip.h"
+#include "rsi_driver.h"
+#include "rsi_m4.h"
+#ifdef COMMON_FLASH_EN
+#include "rsi_power_save.h"
+#endif
+
+rsi_m4ta_desc_t tx_desc[2];
+rsi_m4ta_desc_t rx_desc[2];
+
+uint8_t btn0 = 1;
+uint8_t btn1 = 1;
+
+void sl_button_on_change(uint8_t btn, uint8_t btnAction);
+
+uint32_t NVIC_GetIRQEnable(IRQn_Type IRQn)
+{
+    return ((NVIC->ICER[((uint32_t)(IRQn) >> 5)] & (1 << ((uint32_t)(IRQn) &0x1F))) ? 1 : 0);
+}
+
+void rsi_assertion(uint16_t assertion_val, const char * string)
+{
+    uint16_t i;
+    if (assertion_val == 0)
+    {
+        for (i = 0; i < strlen(string); i++)
+        {
+#ifdef DEBUG_UART
+#ifdef DEBUG_ASSERTION
+            Board_UARTPutChar(string[i]);
+#endif
+#endif
+        }
+
+        return;
+    }
+    else
+    {
+        for (i = 0; i < strlen(string); i++)
+        {
+#ifdef DEBUG_UART
+#ifdef DEBUG_ASSERTION
+            Board_UARTPutChar(string[i]);
+#endif
+#endif
+        }
+
+        while (1)
+            ;
+    }
+}
+
+void IRQ074_Handler(void)
+{
+    ROM_WL_rsi_m4_interrupt_isr(global_cb_p);
+}
+
+void IRQ021_Handler(void)
+{
+    // TODO: Replace with rsi_delay once that is fixed
+    for (int i = 0; i < 10000; i++)
```

Comment:
Why do you need a wait time in an IRQ handler ? The IRQ should be handled as fast as possible

Reply:
added a small delay since some times the IRQ handler is not hitting due to button debouncing

Reply:
not hit or hit a 2nd time without a new press? If it is the latter and debouncing is indeed the issue, maybe disabling the interrupt and then re-enabling it after the debouncing time would be better than this busy wait.